qcelemental/models/molecule.py

```diff
@@ -0,0 +1,813 @@
+
+    @validator('geometry', pre=True)
+    def must_be_3n(cls, v, values, **kwargs):
+        n = len(values['symbols'])
+        try:
+            v = v.reshape(n, 3)
+            if v.shape != (n, 3):
+                raise ValueError()
+        except (ValueError, AttributeError):
+            raise ValueError("Geometry must be castable to shape (N,3)!")
+        return v
+
+    @validator('masses', 'real')
+    def must_be_n(cls, v, values, **kwargs):
+        n = len(values['symbols'])
+        if len(v) != n:
+            raise ValueError("Masses and Real must be same number of entries as Symbols")
+        return v
+
+    @validator('real', whole=True)
+    def populate_real(cls, v, values, **kwargs):
+        n = values["geometry"].shape[0]
+        if len(v) == 0:
+            v = [True for _ in range(n)]
+        return v
+
+    @validator('fragment_charges', 'fragment_multiplicities')
+    def must_be_n_frag(cls, v, values, **kwargs):
+        if 'fragments' in values:
+            n = values['fragments']
+            if len(v) != n:
+                raise ValueError("Fragment Charges and Fragment Multiplicities"
+                                 " must be same number of entries as Fragments")
+        else:
+            raise ValueError("Cannot have Fragment Charges or Fragment Multiplicities "
+                             "without Fragments")
+        return v
+
+    @validator('connectivity')
+    def min_zero(cls, v):
+        if v < 0:
+            raise ValueError("Connectivity entries must be greater than 0")
+        return v
+
+    @property
+    def hash_fields(self):
+        return ["symbols", "masses", "molecular_charge", "molecular_multiplicity", "real", "geometry", "fragments",
+                "fragment_charges", "fragment_multiplicities", "connectivity"]
+
+    class Config:
+        json_encoders = {
+            np.ndarray: lambda v: v.flatten().tolist(),
+        }
+        allow_mutation = False
+
+    # ==========================
+    # Non-Pydantic API functions
+    # ==========================
+
+    def orient(self):
+        warnings.warn("This function is being depreciated in favor of `orient_molecule` "
+                      "to match the schema. They are currently identical in function",
+                      DeprecationWarning)
+        return self.orient_molecule()
+
+    @property
+    def charge(self):
+        warnings.warn("This property is being depreciated in favor of `molecular_charge` "
+                      "to match the schema. They are currently identical in function",
+                      DeprecationWarning)
+        return self.molecular_charge
+
+    @property
+    def multiplicity(self):
+        warnings.warn("This property is being depreciated in favor of `molecular_multiplicity` "
+                      "to match the schema. They are currently identical in function",
+                      DeprecationWarning)
+        return self.molecular_charge
```

Comment:
We need to detect and nuke these ASAP and get them out of the code. One of the reasons we are building these models.